sklearn/mixture/_base.py

```diff
@@ -76,40 +78,26 @@ def _check_initial_parameters(self, X):
         ----------
         X : array-like of shape (n_samples, n_features)
         """
-        if self.n_components < 1:
-            raise ValueError(
-                "Invalid value for 'n_components': %d "
-                "Estimation requires at least one component"
-                % self.n_components
-            )
+        check_scalar(
+            self.n_components,
+            name="n_components",
+            target_type=numbers.Integral,
+            min_val=1,
+        )
 
-        if self.tol < 0.0:
-            raise ValueError(
-                "Invalid value for 'tol': %.5f "
-                "Tolerance used by the EM must be non-negative"
-                % self.tol
-            )
+        check_scalar(self.tol, name="tol", target_type=numbers.Real, min_val=0.0)
 
-        if self.n_init < 1:
-            raise ValueError(
-                "Invalid value for 'n_init': %d Estimation requires at least one run"
-                % self.n_init
-            )
+        check_scalar(
+            self.n_init, name="n_init", target_type=numbers.Integral, min_val=1
+        )
```

Comment:
This change will modifiy the error message because ``check_scalar`` has a different wording.
It needs to be reflected in this test (`test_gaussian_mixture_attributes`)
```py
    n_init_bad = 0
    gmm = GaussianMixture(n_init=n_init_bad)
    msg = (
        f"Invalid value for 'n_init': {n_init_bad} Estimation requires at least one run"
    )
    with pytest.raises(ValueError, match=msg):
        gmm.fit(X)
```

Reply:
Would this work for that specific case?

```python
    msg = (
        f"n_init == {n_init_bad}, must be >= 1."
    )
```
I tried to use some a more verbose message but it seems that ```check_scalar``` messages are expected to be in the form ```'n_init == 0, must be >= 1.'```

Reply:
Yes that's what is expected

Reply:
Thanks @jeremiedbb. I just added the remaining cases!

On another note, it seems that removing the init_params test reduced coverage and codecov is now not passing. Should we devise a new test for that purpose?